packages/twenty-server/src/engine/modules/workspace/services/workspace.service.ts

```diff
@@ -70,4 +74,106 @@ export class WorkspaceService extends TypeOrmQueryService<Workspace> {
+
+      const nextWorkspace = await this.workspaceRepository.findOneBy({
+        id: userWorkspace.workspaceId,
+      });
+
+      if (nextWorkspace) {
+        await this.userRepository.save({
+          id: user.id,
+          defaultWorkspace: nextWorkspace,
+          updatedAt: new Date().toISOString(),
+        });
+        break;
+      }
+
+      // if no workspaces are valid then we delete the user
+      if (index === filteredUserWorkspaces.length - 1) {
+        await this.userRepository.delete({ id: user.id });
+      }
+    }
+  }
+
+  async removeWorkspaceMember(workspaceId: string, memberId: string) {
+    const dataSourceMetadata =
+      await this.dataSourceService.getLastDataSourceMetadataFromWorkspaceIdOrFail(
+        workspaceId,
+      );
+
+    const workspaceDataSource =
+      await this.typeORMService.connectToDataSource(dataSourceMetadata);
+
+    // using "SELECT *" here because we will need the corresponding members userId later
+    const [workspaceMember] = await workspaceDataSource?.query(
+      `SELECT * FROM ${dataSourceMetadata.schema}."workspaceMember" WHERE "id" = '${memberId}'`,
+    );
+
+    if (!workspaceMember) {
+      throw new NotFoundException('Member not found.');
+    }
+
+    await workspaceDataSource?.query(
+      `DELETE FROM ${dataSourceMetadata.schema}."workspaceMember" WHERE "id" = '${memberId}'`,
+    );
+
+    const workspaceMemberUser = await this.userRepository.findOne({
+      where: {
+        id: workspaceMember.userId,
+      },
+      relations: ['defaultWorkspace'],
+    });
+
+    if (!workspaceMemberUser) {
+      throw new NotFoundException('User not found');
+    }
+
+    const userWorkspaces = await this.userWorkspaceRepository.find({
+      where: { userId: workspaceMemberUser.id },
+      relations: ['workspace'],
+    });
+
+    // We want to check if we the user has signed up to more than one workspace
+    if (userWorkspaces.length > 1) {
+      // We neeed to check if the workspace that its getting removed from is its default workspace, if it is then
+      // change the default workspace to point to the next workspace available.
+      if (workspaceMemberUser.defaultWorkspace.id === workspaceId) {
+        await this.reassignDefaultWorkspace(
+          workspaceId,
+          workspaceMemberUser,
+          userWorkspaces,
+        );
+      }
+      // if its not the default workspace then simply delete the user-workspace mapping
+      await this.userWorkspaceRepository.delete({
+        userId: workspaceMemberUser.id,
+        workspaceId,
+      });
+    } else {
+      await this.userWorkspaceRepository.delete({
+        userId: workspaceMemberUser.id,
+      });
+
+      // After deleting the user-workspace mapping, we have a condition where we have the users default workspace points to a
+      // workspace which it doesnt have access to. So we delete the user.
+      await this.userRepository.delete({ id: workspaceMemberUser.id });
+    }
+
```

Comment:
you need to emit an event `workspaceMember.deleted` in order to update the subscription status
`this.eventEmitter.emit('workspaceMember.created', payload);`
See example in packages/twenty-server/src/engine/modules/user-workspace/user-workspace.service.ts